lib/vtk/commands/socks/setup.rb

```diff
@@ -0,0 +1,326 @@
+# frozen_string_literal: true
+
+require_relative '../../command'
+require 'tty-prompt'
+require 'fileutils'
+require 'erb'
+
+module Vtk
+  module Commands
+    class Socks
+      # Sets up socks access to the VA network
+      class Setup < Vtk::Command
+        PROXY_URL = 'https://raw.githubusercontent.com/department-of-veterans-affairs/va.gov-team/master/' \
+          'scripts/socks/proxy.pac'
+
+        attr_reader :ssh_config_path, :input, :output, :boot_script_path, :ssh_key_path, :prompt, :port, :skip_test
+
+        def initialize(options)
+          @options = options
+          @prompt = TTY::Prompt.new interrupt: :exit
+          @port = options['port'] || '2001'
+          @boot_script_path = options['boot_script_path'] || "#{ENV['HOME']}/Library"
+          @ssh_key_path = options['ssh_key_path'] || "#{ENV['HOME']}/.ssh/id_rsa_vagov"
+          @ssh_config_path = options['ssh_config_path'] || "#{ENV['HOME']}/.ssh/config"
+          @skip_test = options['skip_test'] || false
+
+          super()
+        end
+
+        def execute(input: $stdin, output: $stdout)
+          @input = input
+          @output = output
+
+          check_ssh_key
+          configure_ssh
+          test_ssh_connection unless skip_test
+
+          system_boot_configured = configure_system_boot
+          configure_system_proxy
+
+          test_http_connection if system_boot_configured && !skip_test
+
+          log 'SOCKS setup complete.'
+          log 'Please run `vtk socks on` to start your SOCKS connection.' unless system_boot_configured
+        end
+
+        private
+
+        def check_ssh_key
+          return true if key_exists?
+
+          open_key_access_request
+          exit
+        end
+
+        def key_exists?
+          File.exist? ssh_key_path
+        end
+
+        def open_key_access_request
+          url = 'https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?' \
+            'assignees=&labels=external-request%2C+operations&template=Environment-Access-Request-Template.md&' \
+            'title=Access+for+%5Bindividual%5D'
+          log "Please create an SSH key using `ssh-keygen -f ~/.ssh/id_rsa_vagov`. You'll have to wait for access " \
```

Comment:
I tried to run the `ssh-keygen` as a sub-process here but forwarding the stdin here proved extremely difficult. Try if you may but you've been warned.

Having the user run `ssh-keygen` manually and then re-run setup is easier.

Reply:
Turns out it is possible!